src/go/rpk/pkg/cli/cluster/storage/unmount.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newUnmountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "unmount [TOPIC]",
+		Short: "Unmount a topic from the Redpanda Cluster",
+		Long: `Unmount a topic from the Redpanda cluster and secure it in Tiered
+Storage.
+
+This command performs an operation that:
+1. Rejects all writes to the topic
+2. Flushes data to Tiered Storage
+3. Removes the topic from the cluster
+
+The unmount process ensures data safety and cluster integrity. Every
+acknowledged message before the initiation of unmount is guaranteed to be
+persisted in Tiered Storage, subject to Redpanda's existing data safety
+guarantees. 
```

Comment:
@mattschumpert Do we want this in the rpk docs or would we have it elsewhere? It is important, but I wonder if rpk is the right place for this information.

Reply:
Removed it